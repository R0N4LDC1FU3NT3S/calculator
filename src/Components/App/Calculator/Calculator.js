import React, { Component } from "react"
import "./Calculator.css"
import Display from "./Display/Display"
import Keypad from "./Keypad/Keypad"

class Calculator extends Component {
	state = {
		displayValue: "0", // value to be displayed in <Display />
		numbers: [], // values to be displayed in number <Keys />
		operators: [], // values to be displayed in operator <Keys />
		selectedOperator: "", // operator selected for math operation
		storedValue: "" // stored value to use for math operation
	}

	callOperator = () => {
		console.log("call operation")
	}

	setOperator = () => {
		console.log("set operation")
	}

	updateDisplay = () => {
		console.log("update display")
	}

	render = () => {
		// unpacked the component state by using Object Destructuring
		const { displayValue, numbers, operators } = this.state

		return (
			<div className="calculator-container">
				<Display displayValue={displayValue} />
				<Keypad
					callOperator={this.callOperator}
					numbers={numbers}
					operators={operators}
					setOperator={this.setOperator}
					updateDisplay={this.updateDisplay}
				/>
			</div>
		)
	}
}

export default Calculator
