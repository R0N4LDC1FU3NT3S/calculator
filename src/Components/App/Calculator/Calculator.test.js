import React from "react"
import { shallow } from "enzyme"
import Calculator from "./Calculator"
import Display from "./Display/Display"
import Keypad from "./Keypad/Keypad"

describe("Calculator", () => {
	let wrapper

	beforeEach(() => (wrapper = shallow(<Calculator />)))

	it("#0 - Snapshot", () => expect(wrapper).toMatchSnapshot())

	it("#1 - Render correctly", () => {
		expect(wrapper.find("div").length).toEqual(1)
	})

	it("#2 - Render Display", () => {
		expect(wrapper.containsMatchingElement(<Display displayValue={wrapper.instance().state.displayValue} />)).toEqual(true)
	})

	it("#3 - Render Keypad", () => {
		expect(
			wrapper.containsMatchingElement(
				<Keypad
					callOperator={wrapper.instance().callOperator}
					numbers={wrapper.instance().state.numbers}
					operators={wrapper.instance().state.operators}
					setOperator={wrapper.instance().setOperator}
					updateDisplay={wrapper.instance().updateDisplay}
				/>
			)
		).toEqual(true)
	})
})
