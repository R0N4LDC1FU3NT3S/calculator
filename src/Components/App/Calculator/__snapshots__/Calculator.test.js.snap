// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator #0 - Snapshot 1`] = `ShallowWrapper {}`;
